test/test_helpers/assertions.rb

```diff
@@ -6,7 +6,7 @@ def assert_queue_well_formed(queue, msg=nil)
         queue.send(:key_worker_heartbeats), 0, -1, withscores: true)
 
       assert queue.published?
-      assert queue.exhausted?
+      assert queue.fail_fast_limit_reached? || queue.exhausted?
```

Comment:
How does precedence works here? Is that equivalent to:

```ruby
assert(queue.fail_fast_limit_reached? || queue.exhausted?)
```

or

```ruby
assert(queue.fail_fast_limit_reached?) || queue.exhausted?
```

Anyway, we could use parentheses to disambiguate and make it more readable for humans.